services/venice-server/src/main/java/com/linkedin/venice/listener/grpc/handlers/VeniceServerGrpcHandler.java

```diff
@@ -0,0 +1,41 @@
+package com.linkedin.venice.listener.grpc.handlers;
+
+import com.linkedin.venice.listener.grpc.GrpcHandlerContext;
+import com.linkedin.venice.protocols.VeniceServerResponse;
+import io.grpc.stub.StreamObserver;
+
+
+public abstract class VeniceServerGrpcHandler {
+  protected VeniceServerGrpcHandler next;
+
+  public VeniceServerGrpcHandler() {
+  }
+
+  public void addNextHandler(VeniceServerGrpcHandler next) {
+    VeniceServerGrpcHandler current = this;
+    while (current.next != null) {
```

Comment:
Imo we should loop in `VSGRP::addHandler`. Here we should just to `this.next = next`